src/engine/LocalVocab.h

```diff
@@ -20,16 +20,15 @@ class LocalVocab {
 
   // Prevent accidental copying of a local vocabulary (it can be quite large),
   // but moving it is OK.
-  //
-  // TODO: does the default move do the "right" thing, that is, move the hash
-  // map instead of copying it?
   LocalVocab(const LocalVocab&) = delete;
+  LocalVocab& operator=(const LocalVocab&) = delete;
   LocalVocab(LocalVocab&&) = default;
+  LocalVocab& operator=(LocalVocab&&) = default;
 
-  // Get ID of a word in the local vocabulary. If the word was already
-  // contained, return the already existing ID. If the word was not yet
-  // contained, add it, and return the new ID.
-  Id getIdAndAddIfNotContained(const std::string& word);
+  // Get the index of a word in the local vocabulary. If the word was already
+  // contained, return the already existing index. If the word was not yet
+  // contained, add it, and return the new index.
+  LocalVocabIndex getIndexAndAddIfNotContained(const std::string& word);
```

Comment:
There was a comment here that somehow went missing (probably my fault):

You should also add an overload of this function that takes a `std::string&&`,
s.t. we can save the string-copy when inserting a lot of moved words (might happen in your `SERVICE` implementation.